### PR TITLE
POST | painting

### PR DESCRIPTION
**Pull request by LinaYahya**
Relates #16
- verify user and artist middlewares for protected routes
- create addPainting controller to receive image as a file and request body
- parse files at app.js using express-form-data
- create addPainting query
- validate req.body and file
- create uploadImg function using cloudinary
- test route and attach image at test body request 

Screenshot from 2020-06-20 21-41-39

### CODE REVIEW

File: server/controllers/middlewares/verifyUser.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const { verify } = require('jsonwebtoken');
+
+const verifyArtist = async (req, res, next) => {
+  const decoded = await verify(req.cookies.token, process.env.SECRET_KEY);
```
Comment (ranasobeid95) [inline server/controllers/middlewares/verifyUser.js:4]: `Jwt.verify` is an async callback, not a built-in promise so, you can't use `async/await` with it, and if we want to use promise with it we need to build it using `new promise(reject, resolve)`, or you can directly deal with it using callback as I mentioned in suggesting.
```suggestion
 verify(req.cookies.token, process.env.SECRET_KEY, (err, payload) => {
```

File: server/controllers/paintings/addPainting.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const { addPaintingSchema } = require('../../utils/validation');
+const uploadImg = require('../../utils/uploadImg');
+const { addPaintingQuery } = require('../../database/queries');
+
+exports.addPainting = async (req, res, next) => {
+  try {
+    const { title, description, category, property } = req.body;
+    const { id } = req.user;
+    const { paintingImg } = req.files;
```
Comment (ranasobeid95) [inline server/controllers/paintings/addPainting.js:9]: Here you should check if there `req.user` or `req.files` using `if` statement before destructuring
```suggestion
    const { id } = req.user;
    const { paintingImg } = req.files;
```

File: server/controllers/paintings/addPainting.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    await addPaintingSchema.validate(
+      { title, description, category, property, paintingImg },
```
Comment (ranasobeid95) [inline server/controllers/paintings/addPainting.js:11]:
```suggestion
      { ...req.body, paintingImg },
```

Comment (LinaYahya) [reply server/controllers/paintings/addPainting.js:11]: The best practice is to identify the values to validate, cause at front side the user could add undesirable value at the request body

**Pull request merged**